tools/system_libs.py

```diff
@@ -934,6 +966,17 @@ class libcxx(NoBCLibrary, CXXLibrary, NoExceptLibrary, MTLibrary):
   ]
 
 
+class libunwind(CXXLibrary, NoExceptLibrary):
+  name = 'libunwind'
+  cflags = ['-Oz', '-D__USING_WASM_EXCEPTIONS__',
+            '-D_LIBUNWIND_DISABLE_VISIBILITY_ANNOTATIONS']
+  src_dir = ['system', 'lib', 'libunwind', 'src']
+  src_files = ['libunwind.cpp', 'Unwind-wasm.cpp']
+
+  def __init__(self, **kwargs):
+    super(libunwind, self).__init__(**kwargs)
```

Comment:
doesn't this only need to be built with wasm exceptions? I think you can do
```
  def can_use(self):
    return shared.Settings.EXCEPTION_HANDLING

  def can_build(self):
    return shared.Settings.WASM_BACKEND
```

Reply:
Can't both function be `return shared.Settings.EXCEPTION_HANDLING`? I don't think we need to build this when not using wasm EH. Changed to
```py
  def can_build(self):
    return super(libunwind, self).can_build() and shared.Settings.WASM_BACKEND and self.eh_mode == 'wasm'

  def can_use(self):
    return super(libunwind, self).can_use() and shared.Settings.WASM_BACKEND and self.eh_mode == 'wasm'
```
PTAL. (I added `super` calls in the same spirit of #10609)

Reply:
I think `can_build` should not care about the eh mode. The use case is someone runs `embuilder build libunwind-wasm libunwind-noexcept`. All embuilder cares about is whether we can build  the library, but not whether the current compilation settings are compatible - one might build several libraries at once that are not compatible with each other, as in that example.

Reply:
Deleted `self.eh_mode == 'wasm'` part in `can_build`.

Reply:
Deleted `self.eh_mode == 'wasm'` part in `can_build` here in `libunwind`. But I think we actually need this condition in `NoExceptLibrary`'s `can_build`, because, when `eh_mode == 'wasm'`, `-fwasm-exceptions` is added in the command, which fastcomp does not even recognize.